Sprintfinity3902/Collision/CollisionDetector.cs

```diff
@@ -53,32 +55,39 @@ public void Initialize(Game1 game)
          * 
          * maybe this should just take in the room instead of each individual list
          */
-        public void CheckCollision(Dictionary<int, IEntity> enemies, List<IBlock> blocks, List<IEntity> items, List<IEntity> linkProj, List<IEntity> enemyProj, List<IDoor> doors, List<IEntity> garbage) {
+        public void CheckCollision(Dictionary<int, IEntity> enemies, List<IBlock> blocks, List<IEntity> items, List<IEntity> linkProj, List<IEntity> enemyProj, List<IDoor> doors, List<IEntity> garbage, IProjectile bombExplosion) {
             DetectLinkDamage(enemies, enemyProj);
             DetectBlockCollision(enemies, blocks, linkProj, enemyProj);
             DetectEnemyDamage(enemies, linkProj, items, garbage);
             DetectItemPickup(items);
-            DetectDoorCollision(enemies, doors, linkProj, enemyProj);
+            DetectDoorCollision(enemies, doors, linkProj, enemyProj, bombExplosion);
         }
-        private void DetectDoorCollision(Dictionary<int, IEntity> enemies, List<IDoor> doors, List<IEntity> linkProj, List<IEntity> enemyProj)
+        private void DetectDoorCollision(Dictionary<int, IEntity> enemies, List<IDoor> doors, List<IEntity> linkProj, List<IEntity> enemyProj, IProjectile bombExplosion)
         {
             Rectangle linkRect = ((IEntity)link).GetBoundingRect();
             Boolean alreadyMoved = false;
-
             foreach (IDoor door in doors)
             {
                 Rectangle doorRect = door.GetBoundingRect();
-                if (doorRect.Intersects(linkRect))
-                {
-                    if (door.DoorDestination != -1)
+                if (!gameInstance.dungeon.changeRoom.Change) {
+                    if (doorRect.Intersects(linkRect))
                     {
-                        // Add more complex logic here.
-                        gameInstance.dungeon.ChangeRoom(door);
-                    }
-                    else
-                    {
-                        side = blockCollision.SideOfCollision(doorRect, linkRect);
-                        blockCollision.ReflectMovingEntity((IEntity)link, side);
+                        if (door.DoorDestination != -1 && door.CurrentState.IsOpen)
```

Comment:
At some point, it might be helpful to move these implementation details out of Collision Detector, and into it's own handler class. Same with the boomerang change below.